lib/ansible/modules/cloud/amazon/ec2_tag.py

```diff
@@ -171,13 +188,15 @@ def main():
         if not tags:
             module.fail_json(msg="tags argument is required when state is absent")
         for (key, value) in set(tags.items()):
-            if (key, value) not in set(tagdict.items()):
+            if (key, value) not in set(tagdict.items()) and not purge_tags:
                 baddict[key] = value
                 if set(baddict) == set(tags):
                     module.exit_json(msg="Nothing to remove here. Move along.", changed=False)
         for (key, value) in set(tags.items()):
             if (key, value) in set(tagdict.items()):
                 dictremove[key] = value
+        if purge_tags:
+            dictremove = {x:None for x in tagdict if x not in tags}
```

Comment:
While you can create dict comprehensions in python 2.6, you don't need to here:

> tags (dict or list) – Either a dictionary containing name/value pairs or a list containing just tag names. If you pass in a dictionary, the values must match the actual tag values or the tag will not be deleted. If you pass in a value of None for the tag value, all tags with that name will be deleted.

```
tag_remove_list = [x for x in tagdict if x not in tags]
```

The 2.6 equivalent, for reference, is 

```
dictremove = dict((x, None) for x in tagdict if x not in tags)
```

Reply:
Or, even nicer without a comprehension: `set(tagdict) - set(tags)`